### PR TITLE
2123 - Memorable Date validation bug

### PR DESCRIPTION
**Pull request by nickjg231**
## Chromatic

https://2123-memorable-date-validation--60f9b557105290003b387cd5.chromatic.com

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2123

- Addressed an issue where error states in the `va-memorable-date` component were predominantly based on the day input's validity.
- Modified the validation logic to treat each date field (day, month, year) independently, ensuring accurate error messages.
- Updated tests
- Introduced the dayTouched, monthTouched, and yearTouched parameters to improve validation clarity and ensure fields are only marked as invalid after user interaction.

## Testing done
- Local testing with storybook
- Chrome, Safari
- Screen reader (Voice over utility - Mac)

## Screenshots
image
image
image

## Acceptance criteria
- [ ] Month, Day, and Year fields show errors independently based on their validity, not based on the day field's value

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by micahchiang — COMMENTED** [review]
This is what I see on Chromatic, which doesn't seem to match up with the screenshots attached to the PR: 

Screenshot from 2023-10-05 12-43-36

Do I need to change one of the prop values to get the month select drop down?

**Comment by micahchiang** [comment]
ohh, this is the USWDS memorable date version. Sorry, I was looking at the wrong thing!

File: packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import { newE2EPage } from '@stencil/core/testing';
 import { axeCheck } from '../../../testing/test-helpers';
 
-const maxYear = new Date().getFullYear() + 100;
```
Comment (micahchiang) [inline packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:4]: Any particular reason this was removed? 

Edit: Lol ignore, I just checked and it isn't even being used anyway :shrug:

**Review by micahchiang — COMMENTED** [review]
Wonder if something is going on with the changes to `validate`. The errors aren't triggering like I would expect them to in the screenshots attached to the PR description:

[mem-date-screencap.webm](https://github.com/department-of-veterans-affairs/component-library/assets/15097156/320203cc-1d86-4878-95f1-71070a76a687)

My understanding of expected behaviour is that we want errors to proc when a field has been touched and left empty, not necessarily just when inputs are invalid. 

If I enter invalid values for day and year, then the errors do trigger, but there's some unexpected behaviour with prior field errors not being removed from the DOM until blur is fired on the subsequent field:

[Screencast from 10-05-2023 01:38:17 PM.webm](https://github.com/department-of-veterans-affairs/component-library/assets/15097156/9d6c40a7-6048-40db-a332-6e714781b93d)

File: packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts (hunk 1 above)
Comment (jamigibbs) [reply packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:4]: 🤔  This was in a test before. I wonder why/when it got removed? It used to be testing max year validation: https://github.com/department-of-veterans-affairs/component-library/pull/595#discussion_r1060971787

Comment (jamigibbs) [reply packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:4]: I linked to a comment to `va-date` but it seems like there was a test for this compnent before too. Still not sure why it was removed before.

Comment (jamigibbs) [reply packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:4]: I see now that it was updated for a translation string so the variable/value is now codified within i18next: https://github.com/department-of-veterans-affairs/component-library/pull/607/files 

Good clean up. Thanks!

**Review by micahchiang — APPROVED** [review]
Original issue is resolved so approving. Just looks like main needs to be merged in.

**Pull request merged**